stdlib/public/core/UnicodeScalar.swift

```diff
@@ -260,9 +260,9 @@ extension UnicodeScalar : CustomStringConvertible, CustomDebugStringConvertible
 
 extension UnicodeScalar : LosslessStringConvertible {
   public init?(_ description: String) {
-    if let v = UInt32(description) where (v < 0xD800 || v > 0xDFFF)
-      && v <= 0x10FFFF {
-      self = UnicodeScalar(v)
+    _precondition(description.unicodeScalars.count == 1)
```

Comment:
If the initalizer is failable, the precondition is misplaced.  (should return nil instead)